src/corecel/sys/ScopedProfiling.cuda.cc

```diff
@@ -132,33 +134,28 @@ bool ScopedProfiling::enable_profiling()
 /*!
  * Activate nvtx profiling with options.
  */
-ScopedProfiling::ScopedProfiling(Input input)
+void ScopedProfiling::activate_(Input const& input) noexcept
```

Comment:
I know we discussed this before (but maybe we didn't reach a conclusion or we ended up avoiding this)? The trailing underscore means "private" whether it's a data member or function member. You could technically have a std::function or other function-like data member that you'd call with `do_something_()`. I added another instance of this in #933, but since we didn't discuss there maybe we decided to avoid this construct and I forgot about that decision... can you dig up the past discussion? We can add the final decision to the development rst.

Reply:
I'll try to look it up.  In the meantime this line of code:
```
   this->activate_(input);
```
is ambiguous on whether the action is on the current object or (`operator()` called) on a data member named `activate_`.  i.e. the 'scope' of the action can not be determine without looking up the declaration (which naming convention are supposed to help avoid :) ).

Reply:
I can't find the conversion.  I remember it did only indirectly talk about public vs private.  Instead it focused on data object/struct (i.e. public) vs active object (i.e. private) and than indeed there was no ambiguity between:
```
struct ABC { int somedata; }; 
ABC value;
```
which then *always* (since there is no member function) as:
```
value.somedata;
```
so it can not be confused with a local variable, so it is okay in this case to not match the `data member` name convention.

As shown [above](https://github.com/celeritas-project/celeritas/pull/975/files#r1355589375), the same can not be said (and I don't recall we discuss it) for private data members vs private member function.

The resulting usage can be identical (if the data member's type as an `operator()` for example) but, of course, the implication of the scope of the action is completely different.

I.e. it is in my opinion worth to lexicographically differentiate them (private data member and private member function)

Reply:
I don't understand `the implication on the scope of the action is completely different`, and I think the `SoftSurfaceEq` makes the perfect counterexample for our recommendation:
```c++
    SoftEqual<> soft_eq_;

    bool soft_eq_sq_(real_type a, real_type b) const;
    bool soft_eq_distance_(Real3 const& a, Real3 const& b) const;
```

If we write a SoftEqSquare functor that does the same thing as `soft_eq_sq_`, we'll still be calling `this-> soft_eq_sq_(a, b);`. The "scope" of the action should be no more important than if a member function is const or non-const, or whether it has data that points to global data, etc. I think the convention of "this is private vs public" is the only important one.

Reply:
I think I understand by reading again. But I think the way you're using "scope" is imprecise. In both cases you know the thing being called is private and declared in the class. Anything further would have to be found by digging into the implementation in either case, so I don't see the point in differentiating between function-like member data versus member function.

Reply:
@pcanal  You could use the same argument to say we need to use different naming conventions between constant and mutable member functions, right?

Reply:
> between constant and mutable member functions, right?

You could and some naming conventions do.  The difference is that a data member and member function are of different nature (can't 'easily' change the code from to the other) while the other is an annotation (changed by "just" adding the `const` keyword).

Reply:
OK, that's fair. I think we could go either way on this but I'll concede here. I'll merge this as is but immediately add a new commit to change this instance, the existing one in SoftSurfaceEqual, and any other cases I might find. (I think we add `_impl` to some of the other "private" methods as a convention too, now that I htink about it.)

Reply:
> (I think we add _impl to some of the other "private" methods as a convention too, now that I think about it.)

That would work :)